justfile

```diff
@@ -0,0 +1,176 @@
+
+# run the ldmx-sw tests
+test *ARGS:
+    cd build && denv ctest {{ ARGS }}
+
+# run ldmx-sw with the input configuration script
+[no-cd]
+fire config_py *ARGS:
+    denv fire {{ config_py }} {{ ARGS }}
+
+# initialize a containerized development environment
+init:
+    #!/usr/bin/env sh
+    # while setting the denv_workspace is helpful for other
+    # commands that can assume the denv is already initialized,
+    # we need to unset this environment variable to make sure
+    # the test is done appropriately.
+    # just makes sure this recipe runs from the directory of
+    # the justfile so we know we are in the correct location.
+    unset denv_workspace
+    if denv check --workspace --quiet; then
+      echo "\033[32mWorkspace already initialized.\033[0m"
+      denv config print
+    else
+      denv init --clean-env --name ldmx ldmx/dev:latest ${LDMX_BASE}
+    fi
+
+# check that the necessary programs for running ldmx-sw are present
+check:
+    #!/usr/bin/env sh
+    if ! command -v denv 2>&1 > /dev/null; then
+      echo "\033[31mThe program 'denv' is not present.\033[0m"
+      exit 1
+    else
+      echo "\033[32m'denv' has been found.\033[0m"
+    fi
+    # denv can check for container runners it needs
+    denv check
+
+# confirm(PROMPT) just 1.23
+
+# remove the build and install directories of ldmx-sw
+[confirm("This will remove the build and install directories. Are you sure?")]
+clean:
+    rm -r build install
+
+# format the ldmx-sw source code
+format: format-cpp format-just
+
+# format the C++ source code of ldmx-sw
+format-cpp *ARGS='-i':
+    #!/usr/bin/env sh
+    set -exu
+    format_list=$(mktemp)
+    git ls-tree -r HEAD --name-only | egrep '(\.h|\.cxx)$' > ${format_list}
+    denv clang-format {{ ARGS }} $(cat ${format_list})
+    rm ${format_list}
+
+# format the justfile
+format-just:
+    @just --fmt --unstable --justfile {{ justfile() }}
+
+# shellcheck doesn't have a "apply-formatting" option
+# because it really is more of a tidier (its changes could affect code meaning)
+# so only a check is implemented here
+#  ISSUE: the filter implemented here gets all files that are either executable
+#    or have the '.sh' extension. This includes a python script in TrigScint
+#    and some bash-specific scripts as well. Not sure how to handle them.
+
+# check the scripts for common errors and bugs
+shellcheck:
+    #!/usr/bin/env sh
+    set -exu
+    format_list=$(mktemp)
+    git ls-tree -r HEAD | awk '{ if ($1 == 100755 || $4 ~ /\.sh/) print $4 }' > ${format_list}
+    shellcheck --severity style --shell sh $(cat ${format_list})
+
+# below are the mimics of ldmx <cmd>
+# we could think about removing them if folks are happy with committing to the
+# just-style commands above
+
+# open the ROOT shell within the software environment
+root *ARGS="":
+    denv root {{ ARGS }}
+
+# open a ROOT file with a graphical browser
+rootbrowse FILE:
+    denv rootbrowse {{ FILE }}
```

Comment:
So I did
```
xhost + 127.0.0.1
denv rootbrowse
```
on my Mac and I get 
```
 can't open display ":0"
```
I thought we fixed this, I'm confused.

Reply:
I believe you need to either do `xhost +` (no hostname) or `xhost + localhost` (since localhost isn't always 127.0.0.1). I also thought this was fixed and am uncertain about what could be changing in your environment.

Reply:
If doesnt work with `xhost +` either. But ok, I think we can come back to this in another PR